staging/src/k8s.io/apimachinery/pkg/util/net/interface.go

```diff
@@ -75,26 +96,66 @@ func getRoutes(input io.Reader) ([]Route, error) {
 			Interface:   fields[0],
 			Destination: dest,
 			Gateway:     gw,
+			Family:      familyIPv4,
 		})
 	}
 	return routes, nil
 }
 
-// parseHexToIPv4 takes the hex IP address string from route file and converts it
-// from little endian to big endian for creation of a net.IP address.
-// a net.IP, using big endian ordering.
-func parseHexToIPv4(str string) (net.IP, error) {
+func getIPv6DefaultRoutes(input io.Reader) ([]Route, error) {
+	routes := []Route{}
+	scanner := bufio.NewReader(input)
+	for {
+		line, err := scanner.ReadString('\n')
+		if err == io.EOF {
+			break
+		}
+		fields := strings.Fields(line)
+		dest, err := parseIP(fields[0], familyIPv6)
+		if err != nil {
+			return nil, err
+		}
+		gw, err := parseIP(fields[4], familyIPv6)
+		if err != nil {
+			return nil, err
+		}
+		if !dest.Equal(net.IPv6zero) {
+			continue
+		}
+		if gw.Equal(net.IPv6zero) {
+			continue // loopback
+		}
+		routes = append(routes, Route{
+			Interface:   fields[9],
+			Destination: dest,
+			Gateway:     gw,
+			Family:      familyIPv6,
+		})
+	}
+	return routes, nil
+}
+
+// parseIP takes the hex IP address string from route file and converts it
+// to a net.IP address. For IPv4, the value must be converted to big endian.
+func parseIP(str string, family AddressFamily) (net.IP, error) {
 	if str == "" {
 		return nil, fmt.Errorf("input is nil")
 	}
 	bytes, err := hex.DecodeString(str)
 	if err != nil {
 		return nil, err
 	}
-	if len(bytes) != net.IPv4len {
-		return nil, fmt.Errorf("invalid IPv4 address in route")
+	if family == familyIPv4 {
+		if len(bytes) != net.IPv4len {
+			return nil, fmt.Errorf("invalid IPv4 address in route")
+		}
+		return net.IP([]byte{bytes[3], bytes[2], bytes[1], bytes[0]}), nil
+	}
+	// Must be IPv6
+	if len(bytes) != net.IPv6len {
+		return nil, fmt.Errorf("invalid IPv6 address in route")
 	}
-	return net.IP([]byte{bytes[3], bytes[2], bytes[1], bytes[0]}), nil
+	return net.IP(bytes), nil
```

Comment:
correct. It is stored differently in the IPv4 route file vs the IPv6 route file.